src/module.cpp

```diff
@@ -554,6 +554,15 @@ instruction_ref module::get_parameter(std::string name) const
         return this->end();
 }
 
+void module::rename_parameter(instruction_ref ins, const std::string name)
+{
+    assert(ins->name() == "@param");
+    auto& op      = const_cast<operation&>(ins->get_operator());
+    auto param_op = any_cast<builtin::param>(&op);
+    assert(param_op != nullptr);
+    param_op->parameter = name;
```

Comment:
After you change the name then do `ins->replace(op)` to update the operator in the instruction.